pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/TableViews.java

```diff
@@ -144,11 +148,28 @@ public String getSegmentsStatusDetails(
       @Context HttpHeaders headers)
       throws JsonProcessingException {
     tableName = DatabaseUtils.translateTableName(tableName, headers);
-    TableType tableType = validateTableType(tableTypeStr);
-    TableViews.TableView externalView = getTableState(tableName, TableViews.EXTERNALVIEW, tableType);
-    TableViews.TableView idealStateView = getTableState(tableName, TableViews.IDEALSTATE, tableType);
+    final TableType tableType = validateTableType(tableTypeStr);
+    final TableViews.TableView externalView = getTableState(tableName, TableViews.EXTERNALVIEW, tableType);
+    final TableViews.TableView idealStateView = getTableState(tableName, TableViews.IDEALSTATE, tableType);
+
     List<SegmentStatusInfo> segmentStatusInfoListMap = new ArrayList<>();
     segmentStatusInfoListMap = getSegmentStatuses(externalView, idealStateView);
```

Comment:
Not introduced in this PR, but we should loop over segments in IS instead of EV. We can modify the handling to pass in a set of segments to check, and first filter segments based on lineage before checking their status to reduce overhead